service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/managedIdentity/CreateFederatedIdentityStep.java

```diff
@@ -0,0 +1,243 @@
+      KubernetesClientProvider kubernetesClientProvider,
+      LandingZoneApiDispatch landingZoneApiDispatch,
+      SamService samService,
+      WorkspaceService workspaceService,
+      UUID workspaceId,
+      ResourceDao resourceDao) {
+    this.k8sNamespace = k8sNamespace;
+    this.azureConfig = azureConfig;
+    this.crlService = crlService;
+    this.managedIdentityId = managedIdentityId;
+    this.kubernetesClientProvider = kubernetesClientProvider;
+    this.landingZoneApiDispatch = landingZoneApiDispatch;
+    this.samService = samService;
+    this.workspaceService = workspaceService;
+    this.workspaceId = workspaceId;
+    this.resourceDao = resourceDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    if (FlightUtils.getRequired(
+        context.getWorkingMap(), FEDERATED_IDENTITY_EXISTS, Boolean.class)) {
+      logger.info("Federated identity already exists");
+      return StepResult.getStepResultSuccess();
+    }
+
+    var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
+    ControlledAzureManagedIdentityResource managedIdentityResource =
+        resourceDao
+            .getResource(workspaceId, managedIdentityId)
+            .castByEnum(WsmResourceType.CONTROLLED_AZURE_MANAGED_IDENTITY);
+    var uamiName = managedIdentityResource.getManagedIdentityName();
+
+    final AzureCloudContext azureCloudContext =
+        context
+            .getWorkingMap()
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+    var msiManager = crlService.getMsiManager(azureCloudContext, azureConfig);
+    var containerServiceManager =
+        crlService.getContainerServiceManager(azureCloudContext, azureConfig);
+
+    UUID landingZoneId =
+        landingZoneApiDispatch.getLandingZoneId(
+            bearerToken, workspaceService.getWorkspace(workspaceId));
+    var aksCluster =
+        landingZoneApiDispatch
+            .getSharedKubernetesCluster(bearerToken, landingZoneId)
+            .orElseThrow(() -> new RuntimeException("Shared Kubernetes cluster not found"));
+
+    var aksApi =
+        kubernetesClientProvider.createCoreApiClient(
+            containerServiceManager,
+            azureCloudContext.getAzureResourceGroupId(),
+            aksCluster);
+
+    var oidcIssuer =
+        Optional.ofNullable(
+                containerServiceManager
+                    .kubernetesClusters()
+                    .getByResourceGroup(
+                        azureCloudContext.getAzureResourceGroupId(), getResourceName(aksCluster))
+                    .innerModel()
+                    .oidcIssuerProfile()
+                    .issuerUrl())
+            .orElseThrow(() -> new RuntimeException("OIDC issuer not found"));
+    var uamiClientId =
+        msiManager
+            .identities()
+            .getByResourceGroup(azureCloudContext.getAzureResourceGroupId(), uamiName)
```

Comment:
Should this ARM call have `Optional.ofNullable()…orElseThrow()` like the others?